// sqler/index.js


module.exports = {
	/*
	{ firstname: 'Joe',
	  lastname: 'Dailey',
	  email: 'email',
	  password: 'password',
	  swag_size: 'xs',
	  attend_school: 'on',
	  school: 'Arizona State University',
	  state: 'AL',
	  country: 'US',
	  city: 'town',
	  zip: 'postal',
	  travel: 'true',
	  venmo: 'venmo',
	  food_vegetarian: 'on',
	  food_vegan: 'on',
	  food_glutenfree: 'on',
	  food_pescetarian: 'on',
	  food_dairyfree: 'on',
	  food_kosher: 'on',
	  food_halal: 'on',
	  food_other: 'food',
	  Github: 'github',
	  linkedIn: 'linkedin',
	  Dribble: 'dribble',
	  exposure: 'friend',
	  plan_hardware: 'on',
	  plan_hardware_on: 'stuff',
	  plan_mentee: 'on',
	  plan_mentor: 'on',
	  need_partner: 'on',
	  plan_need_partner_skills: 'things' }
	*/
	createApplications:function( callback ){
		global.db.run('CREATE TABLE "applications" ("firstname" BLOB, "lastname" BLOB, "email" BLOB PRIMARY KEY  NOT NULL  UNIQUE , "password" BLOB NOT NULL , "swag_size" BLOB, "is_18" BOOL, "attend_school" BOOL, "school" BLOB, "state" BLOB, "country" BLOB, "city" BLOB, "zip" INTEGER, "travel" BOOL, "venmo" BLOB, "food_vegetarian" BOOL, "food_vegan" BOOL, "food_glutenfree" BOOL, "food_pescetarian" BOOL, "food_dairyfree" BOOL, "food_kosher" BOOL, "food_halal" BOOL, "food_other" BLOB, "Github" BLOB, "linkedIn" BLOB, "Dribble" BLOB, "exposure" BLOB, "plan_hardware" BOOL, "plan_hardware_on" BLOB, "plan_mentee" BOOL, "plan_mentor" BOOL, "need_partner" BOOL, "plan_need_partner_skills" BLOB, "resume" BLOB);', function(err){
			callback(err);
		});
	},
	insertApplication:function(user, callback){
		global.cryptPassword(user.password, function (err, hash){
			if(err){
				console.log(err);
				callback({err:"hash"});
				return;
			}

			var query = 'INSERT INTO applications ("firstname","lastname","email","password","swag_size","is_18","attend_school","school","state","country","city","zip","travel","venmo","food_vegetarian","food_vegan","food_glutenfree","food_pescetarian","food_dairyfree","food_kosher","food_halal","food_other","Github","linkedIn","Dribble","exposure","plan_hardware","plan_hardware_on","plan_mentee","plan_mentor","need_partner","plan_need_partner_skills","resume")'
			query += ' VALUES ('
				+  ' "' + user.firstname + '"'
				+ ', "' + user.lastname + '"'
				+ ', "' + user.email + '"'
				+ ', "' + hash + '"'
				+ ', "' + user.swag_size + '"'
				+ ', "' + ((user.is_18 === 'on') ? true : false) + '"'
				+ ', "' + ((user.attend_school === 'on') ? true : false) + '"'
				+ ', "' + user.school + '"'
				+ ', "' + user.state + '"'
				+ ', "' + user.country + '"'
				+ ', "' + user.city + '"'
				+ ', "' + user.zip + '"'
				+ ', "' + ((user.travel === 'on') ? true : false) + '"'
				+ ', "' + user.venmo + '"'
				+ ', "' + ((user.food_vegetarian === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_vegan === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_glutenfree === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_pescetarian === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_dairyfree === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_kosher === 'on') ? true : false) + '"'
				+ ', "' + ((user.food_halal === 'on') ? true : false) + '"'
				+ ', "' + user.food_other + '"'
				+ ', "' + user.Github + '"'
				+ ', "' + user.linkedIn + '"'
				+ ', "' + user.Dribble + '"'
				+ ', "' + user.exposure + '"'
				+ ', "' + ((user.plan_hardware === 'on') ? true : false) + '"'
				+ ', "' + user.plan_hardware_on + '"'
				+ ', "' + ((user.plan_mentee === 'on') ? true : false) + '"'
				+ ', "' + ((user.plan_mentor === 'on') ? true : false) + '"'
				+ ', "' + ((user.need_partner === 'on') ? true : false) + '"'
				+ ', "' + user.plan_need_partner_skills + '"'
				+ ', "' + user.resume + '"'
				+ ');';
			global.db.run(query, function(err) {
				if(err){
					console.log(err);
					if(err.errno == 19)
						callback({err:"exists"})
					else
						callback({err:"sql"});
				}else{
					callback(null);
				}
			});
		});
	},
	createAnon: function(callback){
		global.db.run('CREATE TABLE "anon" ("state" BLOB, "age" BLOB, "degree" BLOB, "education" BLOB, "gender" BLOB, "ethnicity" BLOB, "parent_education" BLOB, "background" BLOB);', function(err){
			callback(err);
		});
	},
	insertAnon: function(data, callback){
		var query = 'INSERT INTO anon ("state","age","degree","education","gender","ethnicity","parent_education","background")'
			query += ' VALUES ('
				+  ' "' + data.state + '"'
				+ ', "' + data.age + '"'
				+ ', "' + data.degree + '"'
				+ ', "' + data.education + '"'
				+ ', "' + data.gender + '"'
				+ ', "' + data.ethnicity + '"'
				+ ', "' + data.parent_education + '"'
				+ ', "' + data.background + '"'
				+ ');';
		global.db.run(query, function(err) {
			if(err)
				console.log(err);
			callback(err);
		});
	}
};
